tools/tsp-client/src/gh.ts

```diff
@@ -0,0 +1,97 @@
+import { execSync, spawn } from "child_process";
+import { doesFileExist } from "./network.js";
+import { removeDirectory } from "./fs.js";
+import { mkdir } from "fs/promises";
+import { Logger } from "./log.js";
+
+export function getRepoRoot(): string {
+    return execSync('git rev-parse --show-toplevel').toString().trim();
```

Comment:
never use a `sync` method -- it blocks the whole process for no good reason now that we can await things and have readable code.

Reply:
also we should cache this